### PR TITLE
Add support for ElementInternals.role, ariaLabel, and ariaRoleDescription

### PR DESCRIPTION
**Pull request by rniwa**
#### f2257d53ead3d182ed4305f8dc7da6004df12c26
<pre>
Add support for ElementInternals.role, ariaLabel, and ariaRoleDescription
<a href="https://bugs.webkit.org/show_bug.cgi?id=244779">https://bugs.webkit.org/show_bug.cgi?id=244779</a>

Reviewed by Manuel Rego Casasnovas.

Add a runtime flag, DefaultARIAForCustomElementsEnabled, for default ARIA for custom elements,
and add the support for ElementInternals.role, ariaLabel, and ariaRoleDescription. The values of
these default ARIA attributes are stored in CustomElementDefaultARIA via ElementRareData.

* LayoutTests/accessibility/custom-elements/role-expected.txt: Added.
* LayoutTests/accessibility/custom-elements/role.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/form-associated/ElementInternals-accessibility-expected.txt:
* LayoutTests/platform/mac-wk1/TestExpectations:
* LayoutTests/platform/win/TestExpectations:

* Source/WTF/Scripts/Preferences/WebPreferencesExperimental.yaml:
* Source/WebCore/Sources.txt:

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::getAttribute const): Added the support for reading values off of
CustomElementDefaultARIA when there is no content attribute specified.

* Source/WebCore/dom/CustomElementDefaultARIA.cpp: Added.
(WebCore::CustomElementDefaultARIA::setValueForAttribute): Added.
(WebCore::CustomElementDefaultARIA::valueForAttribute const): Added.
* Source/WebCore/dom/CustomElementDefaultARIA.h: Added.

* Source/WebCore/dom/Element.cpp:
(WebCore::Element::customElementDefaultARIA): Added.
(WebCore::Element::customElementDefaultARIAIfExists): Added.
* Source/WebCore/dom/Element.h:

* Source/WebCore/dom/ElementInternals.cpp:
(WebCore::ElementInternals::setRole): Added.
(WebCore::ElementInternals::role): Added.
(WebCore::ElementInternals::setAriaRoleDescription): Added.
(WebCore::ElementInternals::ariaRoleDescription): Added.
(WebCore::ElementInternals::setAriaLabel): Added.
(WebCore::ElementInternals::ariaLabel): Added.
(WebCore::ElementInternals::setAriaValueForAttribute): Added.
(WebCore::ElementInternals::ariaValueForAttribute): Added.
* Source/WebCore/dom/ElementInternals.h:

* Source/WebCore/dom/ElementInternals.idl:

* Source/WebCore/dom/ElementRareData.cpp:
* Source/WebCore/dom/ElementRareData.h:
(WebCore::ElementRareData::customElementDefaultARIA): Added.
(WebCore::ElementRareData::setCustomElementDefaultARIA): Added.
(WebCore::ElementRareData::useTypes const):

* Source/WebCore/dom/Node.cpp:
(WebCore::stringForRareDataUseType):
* Source/WebCore/dom/NodeRareData.h:

Canonical link: <a href="https://commits.webkit.org/254278@main">https://commits.webkit.org/254278@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/4897c0c7177bb641e9d7a64d757fbd8fb16f0fbd

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/88565 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/33128 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/19446 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/97762 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/153476 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/92562 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/31619 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/27200 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/80796 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/92402 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/94195 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/25076 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/75500 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/25034 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/79961 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/80047 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/67997 "Found 1 new API test failure: /TestWTF:WTF_WordLock.ManyContendedLongSections (failure)") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/80285 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/29260 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/14042 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/74056 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/29154 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/15057 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/26229 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/32521 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/37987 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/76912 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/1231 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/31275 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34166 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/17054 "Passed tests") |

### CODE REVIEW

File: Source/WebCore/dom/NodeRareData.h [hunk 1]
```diff
@@ -259,18 +259,19 @@ class NodeRareData {
         ClassList = 1 << 7,
         ShadowRoot = 1 << 8,
         CustomElementQueue = 1 << 9,
```
Comment (mrego) [inline Source/WebCore/dom/NodeRareData.h]: Shouldn't this be renamed to `CustomElementReactionQueue`?

**Review by mrego — APPROVED** [review]
r=me, just leave one small comment inline.

File: Source/WebCore/dom/NodeRareData.h (hunk 1 above)
Comment (rniwa) [reply Source/WebCore/dom/NodeRareData.h]: Oh oops, I missed that. Will fix.

**Comment by rniwa** [comment]
Thanks for the review!

**Pull request merged**